integration-tests/bats/sql-local-remote.bats

```diff
@@ -1033,3 +1033,16 @@ SQL
 
   [[ "$localCherryPickOutput" == "$remoteCherryPickOutput" ]] || false
 }
+
+@test "sql-local-remote: verify checkout will fail early when a server is running" {
+  cd altDB
+  dolt reset --hard
```

Comment:
Is this `reset` just to make sure the workspace is clean before proceeding? Not too clear why we need this.

Reply:
Added comments in the test.